nexusclient/client.go

```diff
@@ -0,0 +1,725 @@
+	if err != nil {
+		return err
+	}
+	httpReq, err := http.NewRequestWithContext(ctx, "POST", url.String(), nil)
+	if err != nil {
+		return err
+	}
+	if request.Header != nil {
+		httpReq.Header = request.Header.Clone()
+	}
+
+	httpReq.Header.Set(headerUserAgent, UserAgent)
+	response, err := c.Options.HTTPClient.Do(httpReq)
+	if err != nil {
+		return err
+	}
+	defer response.Body.Close()
+	if response.StatusCode != http.StatusAccepted {
+		return c.newUnexpectedResponseError(fmt.Sprintf("unexpected response status: %q", response.Status), response)
+	}
+	return nil
+}
+
+// Input for [nexusclient.Client.StartOperation].
+type StartOperationRequest struct {
+	// Name of the operation to start.
+	Operation string
+	// Callback URL to provide to the handle for receiving async operation completions. Optional.
+	CallbackURL string
+	// Request ID that may be used by the server handler to dedupe this start request.
+	// By default a v4 UUID will be generated by the client.
+	RequestID string
+	// Header to attach to the HTTP request. Optional.
+	Header http.Header
+	// Body of the operation request.
+	// If it is an [io.Closer], the body will be automatically closed by the client.
+	Body io.Reader
+}
+
+// NewJSONStartOperationRequest is shorthand for creating a StartOperationRequest with a JSON body.
+// Marhsals the provided value to JSON using [nexusapi.DefaultMarshaler].
+func NewJSONStartOperationRequest(operation string, v any) (request StartOperationRequest, err error) {
+	req := StartOperationRequest{}
+	var b []byte
+	b, err = nexusapi.DefaultMarshaler(v)
+	if err != nil {
+		return
+	}
+	req.Operation = operation
+	req.Header = http.Header{nexusapi.HeaderContentType: []string{nexusapi.ContentTypeJSON}}
+	req.Body = bytes.NewReader(b)
+	return
+}
+
+// StartOperation calls the configured Nexus endpoint to start an operation.
+// The operation may complete synchronously, delivering the result inline, or asynchronously returning a reference that
+// can be used via the returned handle to interact with the operation.
+//
+// Use the returned [OperationHandle] to retrieve the operation's result.
+func (c *Client) StartOperation(ctx context.Context, request StartOperationRequest) (*OperationHandle, error) {
+	if closer, ok := request.Body.(io.Closer); ok {
+		// Close the request body in case we error before sending the HTTP request.
+		defer closer.Close()
+	}
+	url, err := c.joinURL(request.Operation)
+	if err != nil {
+		return nil, err
+	}
+	if request.CallbackURL != "" {
+		q := url.Query()
+		q.Set(nexusapi.QueryCallbackURL, request.CallbackURL)
+		url.RawQuery = q.Encode()
+	}
+	httpReq, err := http.NewRequestWithContext(ctx, "POST", url.String(), request.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if request.Header != nil {
+		httpReq.Header = request.Header.Clone()
+	}
+	if request.RequestID == "" {
+		requestIDFromHeader := request.Header.Get(nexusapi.HeaderRequestID)
```

Comment:
I don't think we need to respect a user manually setting this header. I understand there may be proxy ideas, but proxies would be more transparent than using this client I think.